src/metabase/pulse/render/body.clj

```diff
@@ -420,6 +482,105 @@
       [:img {:style (style/style {:display :block :width :100%})
              :src   (:image-src image-bundle)}]]}))
 
+(def default-y-pos
+  "Default positions of the y-axes of multiple and combo graphs.
+  You kind of hope there's only two but here's for the eventuality"
+  (conj (repeat "right")
+        "left"))
+
+(defn- join-series
+  [names colors types row-seqs y-axis-positions]
+  (let [joined (map vector names colors types row-seqs y-axis-positions)]
+    (vec (for [[card-name card-color card-type rows y-axis-position] joined]
+           {:name          card-name
+            :color         card-color
+            :type          card-type
+            :data          rows
+            :yAxisPosition y-axis-position}))))
+
+
+(s/defmethod render :multiple
+  [_ render-type timezone-id card {:keys [viz-settings] :as data}]
+  (let [multi-res     (pu/execute-multi-card card)
+        ;; multi-res gets the other results from the set of multis.
+        ;; we shove cards and data here all together below for uniformity's sake
+        cards         (cons card (map :card multi-res))
+        multi-data    (cons data (map #(get-in % [:result :data]) multi-res))
+        rowfns        (mapv common/graphing-column-row-fns cards multi-data)
+        row-seqs      (map :rows multi-data)
+        row-seqs      (for [[row-seq rowfnpair] (map vector row-seqs rowfns)]
+                        (let [[x-rowfn y-rowfn] rowfnpair]
+                          (map (juxt x-rowfn y-rowfn)
+                               (common/non-nil-rows x-rowfn y-rowfn row-seq))))
+        col-seqs      (map :cols multi-data)
+        first-rowfns  (first rowfns)
+        [x-col y-col] ((juxt (first first-rowfns) (second first-rowfns)) (first col-seqs))
+        labels        (x-and-y-axis-label-info x-col y-col viz-settings)
+        names         (map :name cards)
+        colors        (take (count multi-data) colors)
+        types         (map :display cards)
+        settings      (->ts-viz x-col y-col labels viz-settings)
+        y-pos         (take (count names) default-y-pos)
+        series        (join-series names colors types row-seqs y-pos)
```

Comment:
the join-series here feels really weird. We pass in arbitrary collections but most of them are `(map :name cards)`, `(map :display cards)`, `(map :row multi-data)`. It seems like there is lots of context, objects where each name has a type, etc, and we just turn them all into sequences and then turn those sequences into a map. Makes me nervous.